packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -286,6 +286,11 @@ export type SnapRuntimeData = {
    * A mutex to prevent concurrent state updates.
    */
   stateMutex: Mutex;
+
+  /**
+   * A mutex to prevent concurrent state decryption.
+   */
+  getStateMutex: Mutex;
```

Comment:
Wondering if we could re-use the `stateMutex`

Reply:
It could, but I'm not sure if it should? We can do encryption and decryption at the same time, they don't depend on each other.